activestorage/app/javascript/activestorage/blob_record.js

```diff
@@ -3,10 +3,11 @@ import { getMetaValue } from "./helpers"
 export class BlobRecord {
   constructor(file, checksum, url) {
     this.file = file
+    const content_type = file.type.length ? file.type : "application/octet-stream"
 
     this.attributes = {
       filename: file.name,
-      content_type: file.type,
+      content_type: content_type,
```

Comment:
I *think* we'd be better off not sending a `content_type` at all when `file.type` is blank and letting Active Storage extract one for us: 
https://github.com/rails/rails/blob/be40e8e5afc6bcf6092392741eec7dcd02edd14a/activestorage/app/models/active_storage/blob.rb#L208
Mind giving that a try?

Reply:
The problem is the Content-Type we get back in the response from the the direct_uploads controller, which is then sent to `/rails/active_storage/disk`.

I tried the following (separately):

```
# blob_record.js:7
this.attributes = {
    filename: file.name,
    byte_size: file.size,
    checksum: checksum
  }

if (file.type) {
  this.attributes.content_type = file.type;
}

# resulted in Mime::Type::InvalidMimeType: "null" is not a valid MIME type
```

```
# blob_upload.js:11

for (const key in headers) {
  this.xhr.setRequestHeader(key, headers[key] || null)
}

# resulted in Mime::Type::InvalidMimeType: "null" is not a valid MIME type
```

```
# blob_upload.js:11

for (const key in headers) {
  if (!headers[key) continue;
  this.xhr.setRequestHeader(key, headers[key])
}

# sends Content-Type over as html and results in 422
```

@javan